docs/markdown/IDE-integration.md

```diff
@@ -93,6 +93,7 @@ can provide code completion for all source files.
 ```json
 {
     "language": "language ID",
+    "machine": "build|host",
```

Comment:
Shouldn't the be `"build|host|unknown"` given the snippet>

Although I wonder if it makes more sense to be `"build" | "host" | null`, since the place where it would be unknown is for custom_target, right?

Reply:
Right, null or absent makes more sense since (unlike language) the machine is defined even in the introspect-from-meson.build case.

Reply:
I think I have a slight preference for null, since technically some custom_targets are wrappers around configure/make and thus could have a machine?, but absent would be fine

Reply:
Right, but I was going for absent exactly because of cases where you do have a machine but Meson cannot place it in the introspection data. Absent leaves the door open for future extension. The "new in 1.7.0" is also missing here in the docs.